bin/publish.ls

```diff
@@ -0,0 +1,150 @@
+require! {
+  'chalk' : {bold, red, cyan, yellow, green, dim}
+  'inquirer'
+  'jsonfile'
+  'semver'
+  'shelljs/global'
+}
+
+
+SUBPROJECTS_TO_PUBLISH =
+  'exocom-mock-javascript'
+  'exocom-server'
+  'exorelay-javascript'
+  'exoservice-javascript'
+
+# exocom-server is used because there is no root package.json
+# all node exocommunication services are in lock-step versioning
+{name, version} = jsonfile.readFileSync './exocom-server/package.json'
+
+if process.argv.length != 3
+  display-help!
+
+level = process.argv[2]
+display-help! if level in ['-h, --help']
+
+target-version = get-target-version version, level
+
+confirm-target-version ->
+  ensure-no-open-files!
+  ensure-on-master!
+
+  check-npm-dependencies!
+  build-subprojects!
+  run-tests!
+  bump-version-numbers!
+  push-version-numbers!
+  publish-to-npm!
+  push-exocom-docker-image!
+  push-git-tags!
+
+
+function check-npm-dependencies
+  console.log green "Checking npm dependencies...\n"
+  run-command-in-subdirs do
```

Comment:
I wanted to make it clear which subdirectory the command was being executed in, so that the output doesn't just look like:
```
Bumping version numbers...

v0.21.3
v0.21.3
v0.21.3
v0.21.3
...
```

The directory names are only known in `run-command-in-subdirs`, however, which is why I was I was building a print statement in two parts.